### DISCUSSION TITLE
GITHUB_TOKEN for Github NPM registry doesn't work inside Docker Build

### DISCUSSION THREAD

**Discussion by pikaju** [community/community#26778] [Packages]
<p>I have a Dockerfile which is supposed to pull a private NPM package from the GitHub NPM package registry. For authentication, I pass the token as a build argument to <code>docker build</code>, and echo it to the <code>.npmrc</code> inside the Dockerfile. I also have GitHub Actions set up using <code>docker/build-push-action@v2</code>.</p>
<ul>
<li>If I pass my personal GitHub token to the Dockerfile, it works just as expected.</li>
<li>If I pass an invalid token (or none at all), I get a “401 Unauthorized” by the <code>yarn install</code> step, which is also expected.</li>
<li>Unfortunately, if I pass the <code>GITHUB_TOKEN</code> in GitHub Actions to the Docker Build instead, I now get a 404 error:</li>
</ul>
<pre><code class="lang-bash"> &gt; [build  8/10] RUN echo "//npm.pkg.github.com/:_authToken=***" &gt; .npmrc &amp;&amp;   yarn install &amp;&amp;   rm -f .npmrc:
#11 0.612 yarn install v1.22.15
#11 0.709 [1/4] Resolving packages...
#11 0.948 [2/4] Fetching packages...
#11 1.511 error An unexpected error occurred: "https://npm.pkg.github.com/download/[Left out but correct]: Request failed \"404 Not Found\"".
#11 1.511 info If you think this is a bug, please open a bug report with the information provided in "/app/yarn-error.log".
#11 1.511 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
</code></pre>
<p>What can I do? Does the <code>GITHUB_TOKEN</code> not work inside of GitHub’s Docker? Of course, I would prefer not using a personal GitHub token in the CI pipeline.</p>

**Comment by pikaju** [comment, marked as answer]
<p>After a lot more research I discovered that the <code>GITHUB_TOKEN</code> is not yet capable of reading packages published by another repository, and it might be a permission issue. See: <a href="https://github.com/actions/setup-node/issues/49" class="inline-onebox" rel="noopener nofollow ugc">GITHUB_TOKEN does not have access to other private packages · Issue #49 · actions/setup-node · GitHub</a></p>
<p>It seems the only viable solution for now is to create a personal access token for the workflow instead.</p>

**Comment by michaelchiche** [comment]
You can access packages by allowing other repository directly on the package settings.
But it is weird, I can make it work directly in the Github Action, but not in the dockerfile...
